File: python/ProgramStudi.py
# Membuat class bernama ProgramStudi
# Class ini merupakan composition class dari Class Dosen, Class Mahasiswa, Class Course

class ProgramStudi:
    
    nama_prodi = "" # Membuat variabel untuk Nama prodi
    kode = ""       # Membuat variabel untuk Kode prodi
    dosen = []      # Membuat variabel untuk dosen dan menyimpan list objek tersebut
    mahasiswa = []  # Membuat variabel untuk mahasiswa dan menyimpan list objek tersebut
    course = []     # Membuat variabel untuk course dan menyimpan list objek tersebut
    
    # Konstruktor kelas ProgramStudi
    def __init__(self, nama_prodi, kode):
        self.nama_prodi = nama_prodi    # Menginisialisasi variabel Nama prodi
        self.kode = kode                # Menginisialisasi variabel Kode prodi
    
    # Getter dan Setter untuk Nama_prodi
    def get_Nama_prodi(self):
        return self.nama_prodi
    
    def set_Nama_prodi(self, nama_prodi):
        self.nama_prodi = nama_prodi
        
    # Getter dan Setter untuk Dosen
    def get_Dosen(self):
        return self.dosen
    
    def set_Dosen(self, dosen):
        self.dosen = dosen
    
    # Getter dan Setter untuk Mahasiswa
    def get_Mahasiswa(self):
        return self.mahasiswa
    
    def set_Mahasiswa(self, mahasiswa):
        self.mahasiswa = mahasiswa
        
    # Getter dan Setter untuk Course
    def get_Course(self):
        return self.course
    
    def set_Course(self, course):
        self.course = course
                
    # Getter dan Setter untuk kode
    def get_Kode(self):
        return self.kode
    
    def set_Kode(self, kode):
        self.kode = kode

    # Method untuk menambahkan dosen ke dalam list dosen
    def tambahDosen(self, dosen):
        self.dosen.append(dosen)
    
    # Method untuk menambahkan mahasiswa ke dalam list mahasiswa
    def tambahMahasiswa(self, mahasiswa):
        self.mahasiswa.append(mahasiswa)
    
    # Method untuk menambahkan course ke dalam list course
    def tambahCourse(self, course):
        self.course.append(course)

File: cpp/ProgramStudi.cpp
class ProgramStudi{
    
    /*
    Saya Talitha Syahla NIM 2101330 mengerjakan Soal Latihan 4 
    dalam mata kuliah Desain Pemrograman Berorientasi Objek 
    untuk keberkahanNya maka saya tidak melakukan kecurangan 
    seperti yang telah dispesifikasikan. Aamiin.
    */

    private:
        string nama_prodi;
        string kode;
        vector<Dosen> dosen;
        vector<Mahasiswa> mahasiswa;
        vector<Course> course; 

    public:
    // Constructor
        ProgramStudi(){
            this->nama_prodi = "";
            this->kode = "";
        }

        ProgramStudi(string nama_prodi, string kode, vector<Dosen> dosen, vector<Mahasiswa> mahasiswa, vector<Course> course){
            this->nama_prodi = nama_prodi;
            this->kode = kode;
            this->dosen = dosen;
            this->mahasiswa = mahasiswa;
            this->course = course;
        }

    // Setter dan Getter

        // Untuk nama_prodi
        string get_Nama_prodi(){
            return this->nama_prodi;
        }

        void set_Nama_prodi(string nama_prodi){
            this->nama_prodi = nama_prodi;
        }

        // Untuk kode
        string get_Kode(){
            return this->kode;
        }

        void set_Kode(string kode){
            this->kode = kode;
        }

        // Untuk Dosen
        vector<Dosen> get_Dosen() {
            return this->dosen;
        }

        void addDosen(Dosen dosen) {
            this->dosen.push_back(dosen);
        }

        // Untuk Mahasiswa
        vector<Mahasiswa> get_Mahasiswa() {
            return this->mahasiswa;
        }

        void addMahasiswa(Mahasiswa mahasiswa) {
            this->mahasiswa.push_back(mahasiswa);
        }
        
        // Untuk Course
        vector<Course> get_Course() {
            return this->course;
        }

        void addCourse(Course course) {
            this->course.push_back(course);
        }

        // Destruktor
        ~ProgramStudi(){
        }
    
};